src/blueprint/blueprint.rs

```diff
@@ -210,4 +214,29 @@ impl Blueprint {
     // generation of schema cannot fail.
     schema.finish().unwrap()
   }
+
+  /// Generate a dumb schema useful for validation.
+  pub fn to_validation_schema(&self) -> Schema {
+    create_dumb_schema(self)
+      .validation_mode(ValidationMode::Strict)
+      .finish()
+      .unwrap()
+  }
+
+  pub async fn validate_operations(&self, operations: Vec<String>) -> Valid<(), String> {
+    let schema = self.to_validation_schema();
+    let mut execution = vec![];
+
+    for op in operations.iter() {
+      match tokio::fs::read_to_string(op).await {
+        Ok(operation) => {
+          let Response { errors, .. } = schema.execute(&operation).await;
+          execution.push((op, errors.iter().map(|e| e.message.clone()).collect::<Vec<String>>()));
+        }
+        Err(_) => execution.push((op, vec!["Cannot read operation".to_string()])),
+      }
+    }
+
+    Valid::from_iter(execution.iter(), |(op, errors)| Fmt::format_operation(op, errors)).map(|_| ())
```

Comment:
Please, I'm very unfamiliar with the `Valid` struct and still trying to understand it. I don't know if someone can explain me why this results in duplicated errors in the console like this:
```bash
Error: Validation Error
• Operation examples/operations/posts.graphql: 2 errors found
1. Unknown field "userOd" on type "Post". Did you mean "userId", "user"?
2. Unknown field "userAd" on type "Post". Did you mean "userId", "user"? []
• Operation examples/operations/empty.graphql: 1 errors found
1. Cannot read operation []
Validation Error
• Operation examples/operations/posts.graphql: 2 errors found
1. Unknown field "userOd" on type "Post". Did you mean "userId", "user"?
2. Unknown field "userAd" on type "Post". Did you mean "userId", "user"? []
• Operation examples/operations/empty.graphql: 1 errors found
1. Cannot read operation []
```
Thanks

Reply:
You're using `Valid` here and then using `Valid::fail` on `format_operation`. Maybe that's the issue?